core/engine/src/object/builtins/jspromise.rs

```diff
@@ -267,34 +268,35 @@ impl JsPromise {
     /// # Examples
     ///
     /// ```
-    /// # use std::error::Error;
+    /// use std::error::Error;
+    /// # use std::cell::RefCell;
     /// # use boa_engine::{
     /// #    object::builtins::JsPromise,
     /// #    builtins::promise::PromiseState,
     /// #    Context, JsResult, JsValue
     /// # };
-    /// async fn f() -> JsResult<JsValue> {
+    /// async fn f(_: &RefCell<&mut Context>) -> JsResult<JsValue> {
     ///     Ok(JsValue::null())
     /// }
     /// let context = &mut Context::default();
     ///
-    /// let promise = JsPromise::from_future(f(), context);
+    /// let promise = JsPromise::from_future(f, context);
     ///
     /// context.run_jobs();
     ///
     /// assert_eq!(promise.state(), PromiseState::Fulfilled(JsValue::null()));
     /// ```
     ///
     /// [async_fn]: crate::native_function::NativeFunction::from_async_fn
-    pub fn from_future<Fut>(future: Fut, context: &mut Context) -> Self
+    pub fn from_future<F>(f: F, context: &mut Context) -> Self
```

Comment:
Would probably be good to rename the function, since it's not taking a `future` anymore.

Reply:
Actually, now that I think about it, do we want to rewrite this function, or would it be better to add a new `from_async_closure` function instead? Seems useful to have the more straightforward `from_future` for when you have a Rust future that you need to resolve which doesn't depend on Javascript values.